fix: print_exception() missing 2 required positional arguments: 'value' and 'tb' and Enabled concurrency with Gradio with Task_ID generations.

Kryst4lDem0ni4s opened the pull request:
Improved the exception handling in HGraphConfig for API version retrieval.

Cause of error discovered:
The traceback.print_exception function requires two more positional arguments, value and tb, which are not provided in the calls from KG Builder.
i.e:

```
builder = KgBuilder(LLMs().get_chat_llm(), Embeddings().get_embedding(), get_hg_client())
File "C:\Users\Khwaish\.vscode\incubator-hugegraph-ai\hugegraph-llm\src\hugegraph_llm\utils\hugegraph_utils.py", line 39, in get_hg_client
return PyHugeClient(
File "C:\Users\Khwaish\.vscode\incubator-hugegraph-ai\.venv\lib\site-packages\pyhugegraph\client.py", line 59, in __init__
self.cfg = HGraphConfig(ip, port, user, pwd, graph, graphspace, timeout or (0.5, 15.0))
File "<string>", line 11, in __init__
File "C:\Users\Khwaish\.vscode\incubator-hugegraph-ai\.venv\lib\site-packages\pyhugegraph\utils\huge_config.py", line 66, in __post_init__
traceback.print_exception(e)
```

imbajin commented:
Thanks, the CI error may relate to here: [`8c7dbaf` (#179)](https://github.com/apache/incubator-hugegraph-ai/pull/179/commits/8c7dbafacc1992f98262db2603666db87576f054)

Kryst4lDem0ni4s commented:
Changed:
```
log_path = "logs/llm-server.log"
    try:
        with open(log_path, "r", encoding='utf-8') as f:
            return ''.join(deque(f, maxlen=lines))
    except FileNotFoundError:
        log.critical("Log file not found: %s", log_path)
        return "LLM Server log file not found."
```
to
```
log_path = "logs/llm-server.log"
    try:
        with open(log_path, "r", encoding='utf-8') as f:
            data = f.read()
            return ''.join(deque(data, maxlen=lines))
    except FileNotFoundError:
        print(f"Log file not found: {log_path}")
        return "LLM Server log file not found."
    except UnicodeDecodeError as e:
        print(f"Decoding error occurred: {e}")
        # ignore the bad bytes and continue processing
        with open(log_path, "r", encoding='utf-8', errors='ignore') as f:
            data = f.read()
            return ''.join(deque(data, maxlen=lines))
```
in order to handle frequent unicode encoding errors during attempts to read logs.

Also added task ID generation for RAG functions, and enabled concurrency using Gradio Blocks in rag_block.py in rag_demo by creating a task ID generation decorator and used Gradio Blocks for concurrency control. Added logs for clear understanding. Issue: #176 https://github.com/apache/incubator-hugegraph-ai/issues/176
